modelforge/train/training.py

```diff
@@ -426,16 +436,22 @@ def _get_forces(
 
         # Compute the gradient (forces) from the predicted energies
         grad = torch.autograd.grad(
-            per_molecule_energy_predict.sum(),
+            per_molecule_energy_predict,
             nnp_input.positions,
-            create_graph=True,
-            retain_graph=True,
+            grad_outputs=torch.ones_like(per_molecule_energy_predict),
+            create_graph=train_mode,
+            retain_graph=train_mode,
+            allow_unused=True,
         )[0]
+
+        if grad is None:
+            raise RuntimeWarning("Force calculation did not return a gradient")
+
         per_atom_force_predict = -1 * grad  # Forces are the negative gradient of energy
 
         return {
             "per_atom_force_true": per_atom_force_true,
-            "per_atom_force_predict": per_atom_force_predict,
+            "per_atom_force_predict": per_atom_force_predict.contiguous(),
```

Comment:
Why are you setting this to be contiguous here? I'm not seeing a this data used in a function that would require contiguous memory allocation but certain could be missing that.

Reply:
it is required by the Pytorch Lightning multi GPU training strategy, which makes assumptions about the memory layout of tensors (which might become non-contiguous if certain operations (view, permutate) are applied to the tensor.